src/Features/Core/Portable/Completion/Providers/Snippets/SnippetCompletionItem.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.Completion.Providers.Snippets
+{
+    internal class SnippetCompletionItem
+    {
+        public static CompletionItem Create(
+            string displayText,
+            string displayTextSuffix,
+            int line,
+            SyntaxToken token,
```

Comment:
this seems to be very C#/VB specific.  How would other roslyn languages do snippets?

Reply:
why do you need to pass a token into this?  what does this token represent?

Reply:
Is there a reason we care about other languages in our snippet implementation? Especially if we're wanting "semantics" to go with our snippets there may not be value in sharing that abstraction. Instead they could rely on the completion item abstraction we already provide.

Reply:
poke @CyrusNajmabadi